gammapy/spectrum/tests/test_dataset.py

```diff
@@ -212,10 +261,22 @@ def test_total_stats(self):
             livetime=self.livetime,
         )
 
-        assert dataset.total_stats.n_on == 4
+        assert dataset.total_stats.n_on == 3
         assert dataset.total_stats.n_off == 40
-        assert dataset.total_stats.excess == 0
+        assert dataset.total_stats.excess == -1
+
+    def test_set_fit_energy_range(self):
+        self.dataset.set_fit_energy_range(emin=0.3*u.TeV, emax=6*u.TeV)
+        assert_allclose(np.where(self.dataset.mask_fit)[0][0],1)
```

Comment:
I find the `np.where(self.dataset.mask_fit)[0][0]` complex, and it's still not obvious to me from the test function why it should be `1`. I guess it would become clear if I read the setup fixture, but it doesn't appear in the diff here.

So I suggest to just assert on the relevant bin in `self.dataset.mask_fit` and don't call `np.where`.
Would that test the same thing and be equally easy to understand?
Or is `np.where(self.dataset.mask_fit)[0][0]` somehow better?

Reply:
replaced with explicit mask values